<a href="https://colab.research.google.com/github/sandeep8689/flyrank-ml-internship-2026/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/sandeep8689/flyrank-ml-internship-2026/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

# My Lane

I selected the Content Refresh lane.

The goal of this lane is to identify existing content that should be updated to improve its search performance. Many web pages lose traffic over time because search trends, competition, and user behavior change. By analyzing search and engagement data, machine learning can help identify which content should be refreshed first.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

# Research Question

**Question:**
Which content items should be refreshed to improve search visibility and user engagement?

**Decision:**
Decide whether a content page should be refreshed.

**Action:**
SEO teams update the content, improve keywords, and optimize the page.

**Cost of a wrong recommendation:**
Refreshing a page that does not need updates wastes time and resources. Missing an outdated page may reduce traffic, user engagement, and search rankings.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [ ]:
!git clone https://github.com/sandeep8689/flyrank-ml-internship-2026.git


Cloning into 'flyrank-ml-internship-2026'...
remote: Enumerating objects: 141, done.
remote: Counting objects: 100% (141/141), done.
remote: Compressing objects: 100% (85/85), done.
remote: Total 141 (delta 54), reused 113 (delta 40), pack-reused 0 (from 0)
Receiving objects: 100% (141/141), 1.82 MiB | 10.48 MiB/s, done.
Resolving deltas: 100% (54/54), done.


In [ ]:
%cd /content/flyrank-ml-internship-2026

/content/flyrank-ml-internship-2026


In [ ]:
import os

for root, dirs, files in os.walk("data"):
    for file in files:
        print(os.path.join(root, file))

data/raw/content_refresh_anonymized.csv


In [ ]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Number of rows:", len(df))
print("Number of columns:", len(df.columns))

print("\nAverage CTR:")
print(df["ctr"].mean())

print("\nAverage Search Volume:")
print(df["search_volume"].mean())

df.head()

Number of rows: 30000
Number of columns: 44

Average CTR:
0.5107333333333334

Average Search Volume:
158.88239139909922


,content_id,client_id,search_volume,competition,competition_level,cpc,content_type,main_intent,word_count,char_count,...,char_count_tier,ctr,avg_position,engagement_rate,scroll_rate,ai_traffic_pct,impression_tier,position_tier,trend_direction,trend_pct
0,content_304f48230142,client_f369cb89fc,10.0,0.67,HIGH,2.05,keyword article,transactional,3221.0,20457.0,...,15000-25000,0.76,10.6,5.88,4.55,0.0,good,striking,down,-41.4
1,content_a1fb4e703a9e,client_4e07408562,90.0,0.01,LOW,0.05,keyword article,informational,2481.0,15562.0,...,15000-25000,0.05,20.3,0.00,10.00,0.0,good,page_3_5,down,-57.7
2,content_9aa793d4d895,client_7f2253d7e2,0.0,0.00,LOW,0.00,keyword article,informational,3515.0,23643.0,...,15000-25000,0.09,36.5,0.00,28.57,0.0,good,page_3_5,down,-60.9
3,content_331d6c4de07b,client_19581e27de,10.0,0.00,LOW,0.00,keyword article,commercial,NaN,NaN,...,NaN,0.49,6.2,1.28,3.45,0.0,good,page_1,stable,-13.8
4,content_d99b7a2d90ca,client_3fdba35f04,0.0,0.00,LOW,0.00,keyword article,informational,2803.0,17469.0,...,15000-25000,0.13,44.0,0.00,24.29,0.0,good,page_3_5,down,-34.7


# Quick Look at the Data

The dataset contains **30,000 rows** and **44 columns**.

The average CTR (Click-Through Rate) is **0.5107**.

The average search volume is **158.88**.

These values show that the dataset contains a large amount of search and SEO performance information. This makes it suitable for analyzing which content should be refreshed to improve search performance and user engagement.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

# Careful Words

This analysis does not prove that refreshing content will always improve search performance.

The model will only identify patterns from historical data. Final decisions should still consider business goals and expert review before updating content.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


# Self Check

- ✅ Selected the Content Refresh lane.
- ✅ Defined the research question.
- ✅ Explained the decision and action.
- ✅ Explained the cost of an incorrect prediction.
- ✅ Explored the dataset with real numbers.
- ✅ Used careful language about what the analysis can and cannot claim.